packages/common/src/core/schemas/internal/getCategoryItems.ts

```diff
@@ -5,36 +5,99 @@ import { IUiSchemaComboboxItem } from "../types";
 /**
  * Fetch the categorySchema for all the categories, including
  * the ones with 0 count, then recursively collet all the child
- * categories on the deepest level. Parse into a format that
- * can be consumed by the combobox field
+ * categories on the deepest level.
  *
- * TODO: render nested categories in combobox
+ * @param orgId The organization id
+ * @param hubRequestOptions The hub request options
+ * @returns a _nested_ structure of categories
  */
-export async function getCategoryItems(
+export async function getNestedCategoryItems(
   orgId: string,
   hubRequestOptions: IHubRequestOptions
 ): Promise<IUiSchemaComboboxItem[]> {
   const url = `${hubRequestOptions.portal}/portals/${orgId}/categorySchema`;
   try {
     const { categorySchema } = await request(url, hubRequestOptions);
-    return parseCategories(categorySchema[0].categories, []);
+    return convertCategoryProps(categorySchema[0].categories);
   } catch (e) {
     return [];
   }
 }
 
-function parseCategories(
-  categories: any[],
-  allCategories: any[]
+/**
+ * Fetch the categorySchema for all the categories, including
+ * the ones with 0 count, then recursively collet all the child
+ * categories on the deepest level.
+ *
+ * @param orgId The organization id
+ * @param hubRequestOptions The hub request options
+ * @returns a _flattened_ structure of categories
+ */
+export async function getFlattenedCategoryItems(
+  orgId: string,
+  hubRequestOptions: IHubRequestOptions
+): Promise<IUiSchemaComboboxItem[]> {
+  const nestedCategories = await getNestedCategoryItems(
+    orgId,
+    hubRequestOptions
+  );
+  return flattenCategories(nestedCategories, []);
+}
+
+/**
+ * Recursively flatten the nested categories
+ *
+ * @param nestedCategories The nested categories
+ * @param allCategories The array to store the flattened categories
+ * @param parents The parent categories
+ * @returns a _flattened_ structure of categories
+ */
+function flattenCategories(
```

Comment:
Do you see a use case for these flattened categories? Since we're not using them I say we delete this.